grammar.js

```diff
@@ -687,13 +688,12 @@ class Label {
                 // rets also forwarded so they can be tied off if the goto is replaced.
                 return this.parent.return(labelScope, this.rets.concat([node]), [], scanner);
             }
-        } else if (expression[0] === 'call') {
-            const label = expression[1][1];
-            const labelScope = scope.label(label);
+        } else if (head === 'call') {
+            const [label, ...args] = tail;
+            const labelScope = scope.label(label[1]); // XXX why the [1]?
```

Comment:
This might be more clear with an `assert(label[0] === 'get')` present, since the expectation is that an expression like `foo` will parse into an S-expression AST like `['get', 'foo']`.

Reply:
Hmm, I guess what confused me, is that the branch above is treating `label` there as an atom, where-as here it's deconstructed; but I guess the above branch just passes the s-expr through?